### PR TITLE
f-checkout@v0.169.0 - add age verification

### PR DESCRIPTION
**Pull request by FayeCarter**
### Added
- Age Verification Page.
- Tests to cover changes.

### Removed
- Some eventData values which were added for testing specific case

### Changed
- `f-form-field` version.

(Stacked dropdowns will be replaced by a date picker soonish 🤞 )
image

### CODE REVIEW

File: packages/components/organisms/f-checkout/src/components/Checkout.vue
Comment (charliearlie) [inline packages/components/organisms/f-checkout/src/components/Checkout.vue:379]: Can remove this until my work is merged in. 🙂

**Review by zlatinivanov-je — COMMENTED** [review]
LGTM, just wonder if `input type="date"` was discussed at all... 
Also not clear on date validation, does not seem to be implemented.

**Comment by FayeCarter** [comment]
> LGTM, just wonder if `input type="date"` was discussed at all...
> Also not clear on date validation, does not seem to be implemented.

So I'm aiming to create a date picker as a new component after this is done.
The validations are also set up as another ticket. 👍

File: packages/components/organisms/f-checkout/src/components/AgeVerification.vue
Comment (ashleynolan) [inline packages/components/organisms/f-checkout/src/components/AgeVerification.vue:16]: Are these form field wrapped in a `form` element at any point? (hard to tell when split into components.

If not, should likely change the outer `div` to a `form` ideally so that it's better semantically.

Not sure on the requirements for this form, but would it be beneficial to have this submit to an endpoint? Just wondering if it's fully client-side validated, how easy it would be for people to get around it potentially (but may not even be an issue if that's accepted risk from product).

File: packages/components/organisms/f-checkout/src/components/_tests/AgeVerification.test.js
Comment (ashleynolan) [inline packages/components/organisms/f-checkout/src/components/_tests/AgeVerification.test.js]: Hard to know without context, but why does this get set to `0`? Does this represent no selected value when it's displayed on screen to the user? If so, may be worth adding a bit more description to the test title (especially as the day gets set to `1`?)

File: packages/components/organisms/f-checkout/src/components/_tests/AgeVerification.test.js
Comment (ashleynolan) [inline packages/components/organisms/f-checkout/src/components/_tests/AgeVerification.test.js]: Is this line needed ?

File: packages/components/organisms/f-checkout/src/components/_tests/AgeVerification.test.js
Comment (ashleynolan) [inline packages/components/organisms/f-checkout/src/components/_tests/AgeVerification.test.js:213]: I'm not a test guru by any means, but rather than testing that it calls `updateDateofBirth` (which you can do too), would it not be better to test that the state itself has been updated to the new date? (as that's the desired result)

File: packages/components/organisms/f-checkout/src/components/AgeVerification.vue
Comment (FayeCarter) [reply packages/components/organisms/f-checkout/src/components/AgeVerification.vue:16]: I'll change it to a form, thanks.

The button will make a call to updateCheckout which should allow it to get properly validated.

File: packages/components/organisms/f-checkout/src/components/_tests/AgeVerification.test.js
Comment (FayeCarter) [reply packages/components/organisms/f-checkout/src/components/_tests/AgeVerification.test.js]: Nope, thanks

File: packages/components/organisms/f-checkout/src/components/_tests/AgeVerification.test.js
Comment (FayeCarter) [reply packages/components/organisms/f-checkout/src/components/_tests/AgeVerification.test.js:213]: I can do, but `handleAgeVerifcation` just calls the action in the store that then calls the mutation that updates the state 🤯 .  So I thought you wouldn't test that the state changes because `handleAgeVerifcation` doesn't directly change the state?

File: packages/components/organisms/f-checkout/src/components/_tests/AgeVerification.test.js
Comment (FayeCarter) [reply packages/components/organisms/f-checkout/src/components/_tests/AgeVerification.test.js]: Thanks, I'll add some description. Annoyingly I'm assuming dates are stored as an array, so when you do `getMonth()` January = 0, Feb =1 etc.

Comment (FayeCarter) [reply packages/components/organisms/f-checkout/src/components/_tests/AgeVerification.test.js]: @ashleynolan I've updated the test description and added a comment, Is that any better?

File: packages/components/organisms/f-checkout/src/components/AgeVerification.vue
Comment (ashleynolan) [inline packages/components/organisms/f-checkout/src/components/AgeVerification.vue]:
```suggestion
    margin: spacing(x2) 0;
```

File: packages/components/organisms/f-checkout/test-utils/component-objects/f-checkout-selectors.js
Comment (ashleynolan) [inline packages/components/organisms/f-checkout/test-utils/component-objects/f-checkout-selectors.js]: Super NIT – can `Note` be lowercase, or does it get auto-populated somewhere (could be one to update in a separate PR).

**Review by ashleynolan — COMMENTED** [review]
Couple of super small things, then LGTM

File: packages/components/organisms/f-checkout/src/components/_tests/AgeVerification.test.js
Comment (ashleynolan) [reply packages/components/organisms/f-checkout/src/components/_tests/AgeVerification.test.js]: Looks good 👍

File: packages/components/organisms/f-checkout/test-utils/component-objects/f-checkout-selectors.js
Comment (FayeCarter) [reply packages/components/organisms/f-checkout/test-utils/component-objects/f-checkout-selectors.js]: It can be updated, It's auto-populated but it's just because we've put `name=Note` instead of `name=note` so I'll change it :)

File: packages/components/organisms/f-checkout/src/components/AgeVerification.vue
Comment (matthewhardern) [inline packages/components/organisms/f-checkout/src/components/AgeVerification.vue:90]: Could this give 31 days on 30 day months plus feb is 28 or 29 days?

File: packages/components/organisms/f-checkout/src/components/AgeVerification.vue
Comment (matthewhardern) [inline packages/components/organisms/f-checkout/src/components/AgeVerification.vue:147]: if the above is true this will output the next available that has a 31 day month if its a 30 day month.

Comment (matthewhardern) [reply packages/components/organisms/f-checkout/src/components/AgeVerification.vue:147]: based On outside communication I understand this will be resolved in a later phase and won't be touching production in this state.

File: packages/components/organisms/f-checkout/src/components/AgeVerification.vue
Comment (matthewhardern) [reply packages/components/organisms/f-checkout/src/components/AgeVerification.vue:90]: see other comment

**Review by matthewhardern — APPROVED** [review]

**Review by charliearlie — DISMISSED** [review]

**Review by charliearlie — APPROVED** [review]

**Pull request merged**
